tests/acceptance/01_vars/02_functions/variablesmatching.cf

```diff
@@ -21,25 +25,15 @@ bundle agent test
       "x_vars" slist => variablesmatching("default:init.test_fbeae67f3e347b5e0032302200141131.*", "x");
       "z_vars" slist => variablesmatching("default:init.test_fbeae67f3e347b5e0032302200141131.*", "z");
 
-      "count" int => length(vars);
-      "x_count" int => length(x_vars);
-      "z_count" int => length(z_vars);
+      "fullvars" data => variablesmatching_as_data("default:init.test_b37ce1b03955522848f0a82d0b2b2e21.*");
+      "x_fullvars" data => variablesmatching_as_data("default:init.test_b37ce1b03955522848f0a82d0b2b2e21.*", "x");
+      "z_fullvars" data => variablesmatching_as_data("default:init.test_b37ce1b03955522848f0a82d0b2b2e21.*", "z");
 }
 
 bundle agent check
 {
-  classes:
-      "ok" and => { strcmp("$(test.count)", "3"),
-                    strcmp("$(test.x_count)", "2"),
-                    strcmp("$(test.z_count)", "0") };
```

Comment:
The JSON comparison has this built in: if the count is off, the JSON data will be different. The expected JSON test is much better in that you see variable count, content, and name differences.